src/codons.py

```diff
@@ -36,35 +47,49 @@ def split_codons(dna: str) -> list[str] | None:
 
     """
     # FIXME: Implement the function
-    return []
+    return dna
+
+
 
 
 def translate_codons(codons: list[str]) -> list[str]:
-    """Translate a list of codons (triplets) into their corresponding
-    amino acid sequence.
+    if codons== None:
+        return None
+    codon=[x.upper() for x in codons]#split_codons(dan)]
+    aminoacid_sequence=[key for ele in codon for val, key in CODON_MAP.items() if ele in val]
```

Comment:
This is a *highly* peculiar way of doing this. It is also somewhat slow because you are running through the codon translation table for each element in the input. You don't need this at all, however, so it doesn't much matter if it is strange or not. You can handle the codon lookup much simpler in the loop you use to check if the keys are valid.

```python
for codon in codons:
     codon = codon.upper()
     if codon not in CODON_MAP:
          return None
     aminoacid_sequence.append(CODON_MAP[codon])
```